presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Memo.java

```diff
@@ -244,15 +277,11 @@ public int getGroupCount()
         return groups.size();
     }
 
-    private static final class Group
+    private final class Group
```

Comment:
You can keep the class static and just pass in `logicalPropertiesProvider` in constructor. Or even better, pass in `logicalProperties` so this constructor just just a normal data initialization constructor.

Reply:
Changed to pass in logicalProperties to the constructor.